src/app/pages/products-list/card/card.component.ts

```diff
@@ -1,22 +1,25 @@
-import {Component} from '@angular/core';
-import {productsMock} from '../../../shared/products/products.mock';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {IProduct} from 'src/app/shared/products/product.interface';
 
 @Component({
     selector: 'app-card',
     templateUrl: './card.component.html',
     styleUrls: ['./card.component.css'],
 })
 export class CardComponent {
-    readonly product = productsMock[0];
+    @Input() product: IProduct | undefined;
 
-    onProductBuy(event: Event) {
+    @Output() ProductBuy = new EventEmitter<IProduct>();
```

Comment:
```suggestion
    @Output() readonly productBuy = new EventEmitter<IProduct | undefined>();
```
1. Нейминг свойств класса пишем в `camalCase`, в `PascalCase` пишем нейминг классов)
2. Для идеальности не хватает `readonly`
3. Для полной честности не хватает `undefined` в типизации, т.к. в `emit` кладется `this.product`, а у него интерфейс `IProduct | undefined`

Reply:
понял, исправил. А вот я видел в одном из ваших комментариев к другим МР, что вы сказали вместо всего объекта, можно просто передать его id, но я не совсем понял, как потом это использовать, То есть мы просто будем искать по id объект и переиспользовать CardComponent?

Reply:
Да, все так, то есть мы отправляем id и скорее всего через сервис этот id попадет в корзину, там мы достанем продукт по id и таким образом получим данные по нему

Reply:
То есть `CardComponent` сообщает о покупке своему родителю, `ProductsListComponent`(родитель `CardComponent`) добавляет id продукта в корзину, корзина сохраняет id купленного продукта и пересчитывает сумму корзины